### ISSUE TITLE
Keep-alive invalidating connection during query execution

### ISSUE THREAD

**Issue by Walkeer** [dbeaver/dbeaver#4704]
latest Deaver 5.2.5 on win 8.1 x64 is invalidating connection during query execution, which result in error message:

`SQL Error [12060] [HY000]: [Vertica][JDBC](12060) ResultSet object has been closed.
  ProxySqlException(connectionId:14cadabf-733c-44eb-ba07-97871ccfa3fa, exceptionId:9651abfc-91ca-4c29-a03f-f5e49db9bb19, message:[Vertica][JDBC](12060) ResultSet object has been closed., errorCode:12060, sqlState:HY000)
  [Vertica][JDBC](12060) ResultSet object has been closed.
`

![dbeaver_error_ads](https://user-images.githubusercontent.com/13622485/49146000-75b34580-f301-11e8-939d-bcb917f93b95.png)

The keep alive is set to one minute, while the query execution takes a bit longer, so the keep alive seem to be trying to execute on the active connection, which probably fails every time. This was not happening in older versions.

**Comment by serge-rider** [comment]
Basically parallel statements shouldn't cause any problems for JDBC connections. However some drivers (like recent Vertica driver) do not support this.
I'll add extra check (connection ping query will be executed only if there no other running queries).
Thanks for the bug report.

**Comment by gonchik** [comment]
Just curious to read a some follow-up

**Comment by yuri-sq** [comment]
Now I have the opposite problem. If a query takes more than hour (this is normal for our analytics), the connection may be closed by some network nodes between the client and the server. If I send pings in parallel to the running query, everything is OK. Can it be automated with DBeaver?

**Comment by E1izabeth** [comment]
@yuri-sq you can set Keep Alive setting to some value other than 0 in connection configuration
![image](https://github.com/dbeaver/dbeaver/assets/28875055/b7358a2a-1f03-465f-b25e-d050d46f7fcf)

**Comment by yuri-sq** [comment]
@E1izabeth yes, that's the problem: Keep-Alive doesn't work when a query is running. So I set Keep-Alive setting, but the keep-alive packet aren't sent.

**Comment by yuri-sq** [comment]
I'm working with Sqream, not Vertica.

**Comment by E1izabeth** [comment]
@yuri-sq could you create the new issue?
